backend/pkg/portforward/handler.go

```diff
@@ -193,24 +194,74 @@ func StartPortForward(kubeConfigStore kubeconfig.ContextStore, cache cache.Cache
 	}
 }
 
+func checkPortForwardPermission(clientset *kubernetes.Clientset, namespace, podName string) error {
+	ctx := context.Background()
+
+	// Create a SelfSubjectAccessReview to check permissions
+	ssar := &authv1.SelfSubjectAccessReview{
+		Spec: authv1.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authv1.ResourceAttributes{
+				Namespace:   namespace,
+				Verb:        "create",
+				Group:       "", // core API group
+				Resource:    "pods",
+				Subresource: "portforward",
+				Name:        podName,
+			},
+		},
+	}
+
+	result, err := clientset.AuthorizationV1().SelfSubjectAccessReviews().Create(ctx, ssar, v1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to check permissions: %w", err)
+	}
+
+	if !result.Status.Allowed {
+		reason := "insufficient permissions"
+		if result.Status.Reason != "" {
+			reason = result.Status.Reason
+		}
+
+		return fmt.Errorf("access denied: %s", reason)
+	}
+
+	return nil
+}
+
 // getKubeClientAndConfig prepares Kubernetes clientset and REST config.
 // It takes a kubeconfig context and an optional bearer token.
 // It returns the configured clientset, REST config, or an error if setup fails.
 func getKubeClientAndConfig(kContext *kubeconfig.Context, token string) (*kubernetes.Clientset, *rest.Config, error) {
-	clientset, err := kContext.ClientSetWithToken(token)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create clientset: %w", err)
-	}
-
 	rConf, err := kContext.RESTConfig()
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to get REST config: %w", err)
 	}
 
+	// Check if exec provider is configured (common for EKS, AKS, GKE)
+	if rConf.ExecProvider != nil {
+		logger.Log(logger.LevelInfo, map[string]string{
+			"provider": rConf.ExecProvider.Command,
+		}, nil, "Using exec provider for authentication")
+
+		// Create clientset with exec provider config
+		clientset, err := kubernetes.NewForConfig(rConf)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to create clientset with exec provider: %w", err)
+		}
+
+		return clientset, rConf, nil
+	}
+
+	// For non-exec based auth, use the token if provided
 	if token != "" {
 		rConf.BearerToken = token
 	}
 
+	clientset, err := kubernetes.NewForConfig(rConf)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create clientset: %w", err)
+	}
+
```

Comment:
If the exec auth is used by default the token will be empty for the request and these changes are not required.
You can simply use `kContext.ClientSetWithToken(token)` like before.

https://github.com/kubernetes-sigs/headlamp/blob/main/backend/pkg/kubeconfig/kubeconfig.go#L273

Reply:
fixed